internal/services/conditionalaccess/conditional_access_policy_resource.go

```diff
@@ -568,14 +568,22 @@ func flattenConditionalAccessSessionControls(in *msgraph.ConditionalAccessSessio
 		return []interface{}{}
 	}
 
-	return []interface{}{
-		map[string]interface{}{
-			"application_enforced_restrictions_enabled": in.ApplicationEnforcedRestrictions.IsEnabled,
-			"cloud_app_security_policy":                 in.CloudAppSecurity.CloudAppSecurityType,
-			"sign_in_frequency":                         in.SignInFrequency.Value,
-			"sign_in_frequency_period":                  in.SignInFrequency.Type,
-		},
+	result := make(map[string]interface{})
+
+	if in.ApplicationEnforcedRestrictions != nil {
+		result["application_enforced_restrictions_enabled"] = in.ApplicationEnforcedRestrictions.IsEnabled
+	}
+
+	if in.CloudAppSecurity != nil {
+		result["cloud_app_security_policy"] = in.CloudAppSecurity.CloudAppSecurityType
+	}
+
+	if in.SignInFrequency != nil {
+		result["sign_in_frequency"] = in.SignInFrequency.Value
+		result["sign_in_frequency_period"] = in.SignInFrequency.Type
 	}
+
+	return []interface{}{result}
```

Comment:
This makes sense as the API doesn't seem to be setting these fields at all when they are "disabled". However, since we always want to set values in state for nested properties, even when they are the default values, can we rather do something like this to ensure that values are always set?

```go
func flattenConditionalAccessSessionControls(in *msgraph.ConditionalAccessSessionControls) []interface{} {
	if in == nil {
		return []interface{}{}
	}

	applicationEnforceRestrictions := false
	if in.ApplicationEnforcedRestrictions != nil {
		applicationEnforceRestrictions = *in.ApplicationEnforcedRestrictions.IsEnabled
	}

	cloudAppSecurity := ""
	if in.CloudAppSecurity != nil && in.CloudAppSecurity.CloudAppSecurityType != nil {
		cloudAppSecurity = *in.CloudAppSecurity.CloudAppSecurityType
	}

	signInFrequency := 0
	signInFrequencyPeriod := ""
	if in.SignInFrequency != nil && in.SignInFrequency.Value != nil && in.SignInFrequency.Type != nil {
		signInFrequency = int(*in.SignInFrequency.Value)
		signInFrequencyPeriod = *in.SignInFrequency.Type
	}

	return []interface{}{
		map[string]interface{}{
			"application_enforced_restrictions_enabled": applicationEnforceRestrictions,
			"cloud_app_security_policy":                 cloudAppSecurity,
			"sign_in_frequency":                         signInFrequency,
			"sign_in_frequency_period":                  signInFrequencyPeriod,
		},
	}
}
```

Reply:
Makes sense to me, I've added that verbatim